PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/completion/csharp/AzureFunctionsTimerTriggerCompletionContributor.kt

```diff
@@ -0,0 +1,90 @@
+ * the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.jetbrains.plugins.azure.functions.completion.csharp
+
+import com.cronutils.descriptor.CronDescriptor
+import com.cronutils.parser.CronParser
+import com.intellij.codeInsight.completion.*
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.util.ProcessingContext
+import com.jetbrains.rider.ideaInterop.fileTypes.csharp.CSharpLanguage
+import com.jetbrains.rider.ideaInterop.fileTypes.csharp.lexer.CSharpTokenType
+import org.jetbrains.plugins.azure.functions.helpers.csharp.AzureFunctionsPsiHelper
+import org.jetbrains.plugins.azure.functions.helpers.NCrontabCronDefinition
+import java.util.*
+
+class AzureFunctionsTimerTriggerCompletionContributor : CompletionContributor() {
+
+    private val cronParser = CronParser(NCrontabCronDefinition)
+    private val cronDescriptor = CronDescriptor.instance(Locale.getDefault())
+    private val cronSuggestions = arrayOf(
+            "* * * * * *",
+            "0 * * * * *",
+            "0 */5 * * * *",
+            "0 0 * * * *",
+            "0 0 */6 * * *",
+            "0 0 8-18 * * *",
+            "0 0 0 * * *",
+            "0 0 10 * * *",
+            "0 0 * * * 1-5",
+            "0 0 0 * * 0",
+            "0 0 9 * * Mon",
+            "0 0 0 1 * *",
+            "0 0 0 1 1 *",
+            "0 0 * * * Sun",
+            "0 0 0 * * Sat,Sun",
+            "0 0 0 * * 6,0",
+            "0 0 0 1-7 * Sun",
+            "11 5 23 * * *",
+            "*/15 * * * * *",
+            "0 30 9 * Jan Mon"
+    )
+
+    init {
+        extend(CompletionType.BASIC,
+                PlatformPatterns.psiElement(CSharpTokenType.STRING_LITERAL_REGULAR).withLanguage(CSharpLanguage),
+                object : CompletionProvider<CompletionParameters?>() {
+
+                    override fun addCompletions(
+                            parameters: CompletionParameters,
+                            context: ProcessingContext,
+                            resultSet: CompletionResultSet) {
+
+                        val element = parameters.position
+                        val timerTriggerAttributeElement = AzureFunctionsPsiHelper.tryResolveTimerTriggerAttributeFromScheduleExpressionStringLiteralSibling(element)
+                        if (timerTriggerAttributeElement != null) {
+                            for (cronSuggestion in cronSuggestions) {
+
+                                val cron = cronParser.parse(cronSuggestion)
```

Comment:
Maybe better to handle exception on parse() failure here. Log and continue with other suggestions or throw after processing?

Reply:
Did you get an exception on any of the suggestions? I'll add a check for it, but ideally it should not throw at all with the examples in the file :-)